prometheus-to-sd/translator/translator.go

```diff
@@ -0,0 +1,150 @@
+		if err != nil {
+			glog.Warningf("Error while processing metric %s: %v", name, err)
+		} else {
+			ts = append(ts, t...)
+		}
+	}
+	return ts
+}
+
+func filterWhitelisted(allMetrics map[string]*dto.MetricFamily, whitelisted []string) map[string]*dto.MetricFamily {
+	glog.V(4).Infof("Exporting only whitelisted metrics: %v", whitelisted)
+	res := map[string]*dto.MetricFamily{}
+	for _, w := range whitelisted {
+		if family, found := allMetrics[w]; found {
+			res[w] = family
+		} else {
+			glog.Infof("Whitelisted metric %s not present in Prometheus endpoint.", w)
+		}
+	}
+	return res
+}
+
+func translateFamily(config *config.GceConfig, component string, family *dto.MetricFamily, start, end time.Time) ([]*v3.TimeSeries, error) {
+	glog.V(4).Infof("Translating metric family %v", family.GetName())
+	var ts []*v3.TimeSeries
+	if family.GetType() != dto.MetricType_COUNTER && family.GetType() != dto.MetricType_GAUGE {
+		return ts, fmt.Errorf("Metric type %v of family %s not supported", family.GetType(), family.GetName())
+	}
+	for _, metric := range family.GetMetric() {
+		t := translateOne(config, component, family.GetName(), family.GetType(), metric, start, end)
+		ts = append(ts, t)
+		glog.V(4).Infof("%+v\nMetric: %+v, Interval: %+v, Value: %+v", *t, *(t.Metric), t.Points[0].Interval, *(t.Points[0].Value.Int64Value))
+	}
+	return ts, nil
+}
+
+func translateOne(config *config.GceConfig, component string, name string, mType dto.MetricType, metric *dto.Metric, start, end time.Time) *v3.TimeSeries {
+	fullName := fmt.Sprintf("%s/%s/%s", config.Prefix, component, name)
+
+	var metricKind string
+	var metricValue int64
+	interval := &v3.TimeInterval{
+		EndTime: end.Format(time.RFC3339),
+	}
+	if mType == dto.MetricType_COUNTER {
+		metricKind = "CUMULATIVE"
+		metricValue = int64(metric.GetCounter().GetValue())
+		interval.StartTime = start.Format(time.RFC3339)
+	} else {
+		metricKind = "GAUGE"
+		metricValue = int64(metric.GetGauge().GetValue())
+	}
+
+	point := &v3.Point{
+		Interval: interval,
+		Value: &v3.TypedValue{
+			Int64Value:      &metricValue,
+			ForceSendFields: []string{"Int64Value"},
+		},
+	}
+
+	return &v3.TimeSeries{
+		Metric: &v3.Metric{
+			Labels: getMetricLabels(metric.GetLabel()),
+			Type:   fullName,
+		},
+		Resource: &v3.MonitoredResource{
+			Labels: getResourceLabels(config),
+			Type:   "gke_container",
+		},
+		MetricKind: metricKind,
+		ValueType:  "INT64",
+		Points:     []*v3.Point{point},
+	}
+}
+
+func getMetricLabels(labels []*dto.LabelPair) map[string]string {
+	metricLabels := map[string]string{}
+	for _, label := range labels {
+		metricLabels[label.GetName()] = label.GetValue()
+	}
+	return metricLabels
+}
+
+func getResourceLabels(config *config.GceConfig) map[string]string {
+	return map[string]string{
+		"project_id":     config.Project,
+		"cluster_name":   config.Cluster,
+		"zone":           config.Zone,
+		"instance_id":    config.Instance,
+		"namespace_id":   "",
```

Comment:
I used the same approach that here https://github.com/kubernetes/contrib/blob/4fa96c4b20520248bfdd4db97ff869c7750d5188/kubelet-to-gcm/monitor/controller/translator.go#L63

cc @Q-Lee